tests/test_aggregates.py

```diff
@@ -129,131 +129,144 @@ def test_groupby_with_invalid_length_default_key_for_Iterable(self):
         with pytest.raises(ValueError):
             aggregator.groupby(["half", "even"], default=[-1, -1, -1])
 
-    def test_aggregate_hashing(self, list_of_aggregates):
+    def test_aggregate_hashing(self, list_of_aggregators):
         # Since we need to store groups on a per aggregate basis in the project,
         # we need to be sure that the aggregates are hashing and compared correctly.
         # This test ensures this feature.
-        # list_of_aggregates contains 14 distinct storing objects (because an
+        # list_of_aggregators contains 14 distinct store objects (because an
         # aggregator object is differentiated on the basis of the `_is_aggregate` attribute).
         # When this list is converted to set, then these objects are hashed first
         # and then compared. Since sets don't carry duplicate values, we test
         # whether the length of the set obtained from the list is equal to 14 or not.
-        assert len(set(list_of_aggregates)) == 14
+        assert len(set(list_of_aggregators)) == 14
+        # Ensure that equality implies hash equality.
```

Comment:
@bdice I am unable to understand why do we need this test as python sets use hash values and equality checks (right?)

Reply:
We implement both `__hash__` and `__eq__` for the aggregator class. It is a requirement of the Python data model that `a == b` implies `hash(a) == hash(b)`. Since our implementation of the methods doesn't necessarily guarantee that (if they were implemented incorrectly, for example), I am verifying that manually with this test.

Python sets can contain objects with the same hash as long as they are not equal. Example:

```python
class A:
    def __hash__(self):
        # All instances have the same hash.
        return 42
    def __eq__(self, other):
        # Instances are never equal.
        return False

a1 = A()
a2 = A()
assert len({a1, a2}) == 2
```

Reply:
I now understand this, thanks!